pkg/api/types.go

```diff
@@ -748,30 +748,55 @@ func (p *Properties) K8sOrchestratorName() string {
 	return ""
 }
 
-func (p *Properties) getAgentVMPrefix() string {
-	return p.K8sOrchestratorName() + "-agentpool-" + p.GetClusterID() + "-"
+func (p *Properties) getAgentPoolIndexByName(name string) int {
+	index := -1
+	for i, profile := range p.AgentPoolProfiles {
+		if profile.Name == name {
+			index = i
+			break
+		}
+	}
+	return index
+}
+
+// GetAgentVMPrefix returns the VM prefix for an agentpool
+func (p *Properties) GetAgentVMPrefix(a *AgentPoolProfile) string {
```

Comment:
Let's address the case of -1 index in this method. Return an empty string maybe ? We can add a test to reflect this as well.